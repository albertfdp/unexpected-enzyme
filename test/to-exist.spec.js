import expect from './unexpected-enzyme';
import { mount } from 'enzyme';
import React, { Component } from 'react';

class Fixture extends React.Component {
  render() {
    return (
      <div>
        <div id="parent" />
      </div>
    );
  }
}

const NullFixture = () => null;

describe('to-have-props-satisfying', () => {
  it('passes when the actual matches the expected', () => {
    const wrapper = mount(<Fixture />);

    expect(wrapper.find('#parent'), 'to be present');
    expect(wrapper.find('#parent'), 'to exist');
  });

  it('fails when the actual does not match the expected', () => {
    const wrapper = mount(<Fixture />);

    expect(
      () => expect(wrapper.find('#parent'), 'not to be present'),
      'with error matching snapshot'
    );
  });
});
