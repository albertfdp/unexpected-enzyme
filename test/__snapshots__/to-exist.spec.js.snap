// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to-exist mount fails when the actual does not match the expected 1`] = `
"Element did not exist in:

<div><div id=\\"parent\\"></div></div>"
`;

exports[`to-exist mount fails when the actual negated matches the expected 1`] = `"expected <div id=\\"parent\\"></div> not to exist"`;

exports[`to-exist mount with queried for fails when the actual does not match the expected 1`] = `
"expected <div><div id=\\"parent\\"></div></div> queried for '#foobar' to exist
  Element did not exist in:

  <div><div id=\\"parent\\"></div></div>"
`;

exports[`to-exist shallow fails when the actual does not match the expected 1`] = `
"Element did not exist in:

<div><div id=\\"parent\\" /></div>"
`;

exports[`to-exist shallow fails when the actual negated matches the expected 1`] = `"expected <div id=\\"parent\\" /> not to exist"`;

exports[`to-exist shallow with queried for fails when the actual does not match the expected 1`] = `
"expected <div><div id=\\"parent\\" /></div> queried for '#foobar' to exist
  Element did not exist in:

  <div><div id=\\"parent\\" /></div>"
`;
